src/content/language/dot-shorthand.md

```diff
@@ -0,0 +1,280 @@
+---
+title: Shorthand syntax
+description: Learn about the shorthand syntax in Dart.
+shortTitle: Shorthand
+prevpage:
+  url: /language/enums
+  title: Enums
+nextpage:
+  url: /language/extension-methods
+  title: Extension methods
+---
+
+:::experimental
+Support for shorthand syntax is experimental and
+can only be used on the `main` channel.
+:::
+
+Write more concise Dart code by using the shorthand syntax `.foo` 
+instead of `ContextType.foo`. The compiler can infer the type 
+whenever it's clear from the context, letting you access 
+enum values, static members, or constructors more cleanly.
+
+Here is a quick look:
+
+<?code-excerpt "language/lib/classes/shorthand.dart (intro)"?>
+```dart
+// Enum example
+enum Status { none, running, stopped, paused }
+Status currentStatus = .running; // Instead of Status.running
+
+// Static method example
+int port = .parse('8080'); // Instead of int.parse('8080')
+
+// Constructor example
+class Point {
+  final int x, y;
+  Point(this.x, this.y);
+  Point.origin() : x = 0, y = 0;
+}
+Point origin = .origin(); // Instead of Point.origin()
+```
+## Using shorthand syntax
+
+### Enums
+A primary and highly recommended use case for shorthands is with enums, 
+especially in assignments and switch statements, where the 
+enum type is very clear.
+
+<?code-excerpt "language/lib/classes/shorthand.dart (enums)"?>
+```dart
+// Enum example
+enum LogLevel { debug, info, warning, error }
+
+// Function to get a color code based on log level
+String colorCode(LogLevel level) {
+  return switch (level) {
+    .debug   => 'gray',    // Instead of LogLevel.debug
+    .info    => 'blue',    // Instead of LogLevel.info
+    .warning => 'orange',  // Instead of LogLevel.warning
+    .error   => 'red',     // Instead of LogLevel.error
+  };
+}
+
+// Example usage:
+// var warnColor = colorCode(.warning); // Returns 'orange'
+```
+
+### Constructors
+Shorthands are useful for invoking named constructors, 
+static factory methods, or constructors of generic classes 
+where type arguments are provided.
+
+<?code-excerpt "language/lib/classes/shorthand.dart (constructors)"?>
+```dart
+ class Point {
+  final double x, y;
+  const Point(this.x, this.y);
+  const Point.origin() : x = 0, y = 0; // Named constructor
+  static Point fromList(List<double> list) => Point(list[0], list[1]); // Static method
+}
+
+// Named constructor
+Point origin = .origin(); // Instead of Point.origin()
+
+// Static factory method
+Point p1 = .fromList([1.0, 2.0]); // Instead of Point.fromList([1.0, 2.0])
+
+// With type arguments for generic class constructors
+List<int> intList = .filled(5, 0); // Instead of List<int>.filled(5, 0)
```

Comment:
This isn't the equivalent to the comment of `List<int>.filled(5, 0)` actually. It's equivalent to `List.filled(5,0)` where the compiler then infers the type parameters. It's obvious here, but there are some examples where this makes a difference.

This is a contrived example, but something like this wouldn't work if we pulled the type parameters from the context:
```

class C<T> {
  static C<X> foo<X>(X x) => new C<X>();
  C<U> cast<U>() => new C<U>();
}

void main() {
  C<bool> c = .foo("String").cast(); // C.foo and not C<bool>.foo
}
```